src/core/shape_inference/include/strided_slice_shape_inference.hpp

```diff
@@ -36,7 +37,7 @@ void shape_infer(const StridedSlice* op,
                           end_shape.rank(),
                           ").");
 
-    const auto& strides_shape = input_shapes[3];
+    const auto& strides_shape = input_shapes.size() < 4 ? T{DimType(-1)} : input_shapes[3];
```

Comment:
`DimType` is a `StaticDimension` for the `shape_infer` call with `T` as `StaticShape`, 
it means the value is assigned to `size_t`, so it's not safe to init such dimension (DimType) with `-1`.

Also default `stride` input is created by `StridedSlice` constructor, is that even possible that the shape is not provided?

Anyway it's used only for the validation below (L41), so my suggestion is to remove the `strides_shape` variable and rewrite the check like:
```
    NODE_VALIDATION_CHECK(op,
                          input_shapes.size() < 4 || input_shapes[3].rank().compatible(1),
                          "End input must be 1D (end rank: ",
                          input_shapes[3].rank(),
                          ").");
```

Reply:
There is a case with default strides created but data not provided for `shape_infer` then instead use default stride values the exception will be thrown.

Proposed check will cause exception instead of use default stride value.

Create static shape with `-1` is OK as this value will be casted to size_t (oxff.ff) and created wanted dummy shape which will be tread as shape without items, it could be create with explicit cast 
`T{DimType(static_cast<typename DimType::value_type>(-1))}`

But removed this dummy stride shape as this input is always set in op. When not provided in `input_shapes` use operator's strides input shape.